### ISSUE TITLE
code that checks whether user has set certain settings explicitly or not is broken when pyrightconfig/pyproject exists

### ISSUE THREAD

**Issue by heejaechang** [microsoft/pylance-release#6499]
## Environment data

-   Pylance version: 2024.9.102

Pylance has code to see whether certain settings are set by users explicitly or not when applying different default values for each settings.

currently that logic only checks `vscode's settings.json` but not `pyrightconfig/pyproject` wrongly assuming certain settings are not set explicitly.

issue is found through code but not through set repro steps.

**Comment by rchiodo** [comment]
Are you referring to the logic you added about whether or not a setting is set to default?

**Comment by heejaechang** [comment]
yep. that only checks `vscode's settings` but not `pyright/pyproject` files.

**Comment by heejaechang** [comment]
I revalidated the concrete behavior path this report describes on current bits.

The current executable contract for this settings/config-precedence case now passes in sync, async, and typeServer modes: when `generateWithTypeAnnotation` is still default-derived from client settings and `pyrightconfig.json` sets `typeCheckingMode` to `basic`, override completion still includes type annotations as expected.

Because this report was originally based on code inspection rather than a separate end-user repro, that current contract is the strongest verification artifact for this exact defect. I’m closing this as no longer reproducible on current bits.

If you can still reproduce a user-visible case with a current Pylance build, please open a new issue with the exact setting/config combination and expected vs actual behavior.
